pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -168,6 +169,7 @@ public JavaInstanceRunnable(InstanceConfig instanceConfig,
         this.collectorRegistry = collectorRegistry;
 
         this.instanceClassLoader = Thread.currentThread().getContextClassLoader();
+        ReflectionUtils.setClassLoader(instanceClassLoader);
```

Comment:
I believe that this will not fix the problems we have.

Therefore I discovered this other bad problem about using AVRO (and JsonNode...)
https://github.com/apache/pulsar/issues/11274

basically now (2.8.0) you cannot use the org.apache.avro...GenericRecord returned by GenericRecord.getNativeObject()